jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -516,31 +516,38 @@ public void configure(Object obj, XmlParser.Node cfg, int i) throws Exception
         private void set(Object obj, XmlParser.Node node) throws Exception
         {
             String attr = node.getAttribute("name");
+            String name = "set" + attr.substring(0, 1).toUpperCase(Locale.ENGLISH) + attr.substring(1);
             String id = node.getAttribute("id");
             String property = node.getAttribute("property");
             String propertyValue = null;
+
+            Class<?> oClass = nodeClass(node);
+            if (oClass != null)
+                obj = null;
+            else
+                oClass = obj.getClass();
+
             // Look for a property value
             if (property != null)
             {
                 Map<String, String> properties = _configuration.getProperties();
                 propertyValue = properties.get(property);
                 // If no property value, then do not set
                 if (propertyValue == null)
+                {
+                    // check that there is at least one setter that could have matched
+                    if (Arrays.stream(oClass.getMethods()).noneMatch(m -> m.getName().equals(name)))
+                        throw new NoSuchMethodException(String.format("No '%s' on %s", name, oClass.getName()));
+                    // otherwise it is a noop
```

Comment:
When this method is actually going to set a value, it checks for an exact setter, then a native method, then a field, and then finally other methods that might match.  That being the case, failing if there's no exact setter, even when no value is being set, misses the cases where it could have been a native or field match.

Reply:
I've added a check for a public field.  All the different setter type are covered by the same name, so that is OK.